practice-app/website/templates/base.html

```diff
@@ -29,7 +29,12 @@ <h1>
                 function search() {
                     var form = document.getElementById("searchBar");
                     var formData = new FormData(form)
+                    {% if token %}
+                    url = "/?token={{token}}&query=" + encodeURIComponent(formData.get("query"))
```

Comment:
Showing the token on URL is insecure, instead of this sessions or cookies can be used

Reply:
I have updated the token system to store the tokens in a Flask session instead of the URL.